src/System.Buffers.Primitives/System/Buffers/OutputWriter.cs

```diff
@@ -48,9 +61,23 @@ public void Write(ReadOnlySpan<byte> source)
             }
         }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Ensure(int count = 1)
         {
+            if (_span.Length < count)
+            {
+                EnsureMore(count);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private void EnsureMore(int count = 1)
```

Comment:
count can be 0

Reply:
Does `_output.GetMemory(0)` always return `span.Length >= 1`?

Reply:
Yes, since we don't do `Append` anymore there is no reason to return empty when someone asks for memory.